src/deserialize/parse.rs

```diff
@@ -26,8 +34,13 @@ pub(crate) fn parse_edn(
         Some((_, ':')) => read_key_or_nsmap(chars),
         Some((_, '-')) => Ok(read_number('-', chars)?),
         Some((_, '\\')) => Ok(read_char(chars)?),
+        Some((_, ';')) => {
+            read_comment(chars)?;
+            Ok(parse_edn(chars.next(), chars)?)
```

Comment:
if you call `parse_edn` here you will be missing the full parse (`parse`), as `parse_edn` only parses "primitive" types. Also, I am not sure you actually need this line `Ok(parse_edn(chars.next(), chars)?)`, maybe only if comments come as the first line of code.

Reply:
only parse calls parse_edn, so if parse_edn calls itself, it will still return back to parse?

Yes that is for comments as the first line, e.g. test parse_str_top_level_comment on line 494

Reply:
`parse`  calls `parse_edn`, https://github.com/naomijub/edn-rs/blob/master/src/deserialize/parse.rs#L16.
I think this is the reason why 
> the top level parse doesn't catch inside a map, vec, set, etc., so if i comment this line the following test fails (from line 852): `test deserialize::parse::test::parse_comment_inside_map ... FAILED`
is failing

Reply:
It calls back to top level now.

Reply:
I see what you mean, this shouldn't be needed, I have removed these from parse_edn